### ISSUE TITLE
avoid recency bias in prompt construction

### ISSUE THREAD

**Issue by AndreasKarasenko** [iryna-kondr/scikit-llm#104]
**Context**
According to this [paper](http://proceedings.mlr.press/v139/zhao21c/zhao21c.pdf) ChatGPT (and likely other LLMs) suffer from a recency bias. Whatever class comes last has a higher propability of being selected.
**Issue**
Currently scikit-llm constructs prompts based on the order of the training data. 
Since we are recommended to restrict the training data I would usually do something like this:
~~~python
df = df.groupby(label_col).apply(lambda x: x.sample(n_samples))
df = df.reset_index(drop=True)
~~~
Which returns a sorted dataframe by label_col. Even if `sort=False` is passed to `groupby` the instances are still clustered by label.

**Question/Solution**
Should a method be implemented that randomizes the order of samples in the prompt / training data, or should users take care of that themselves?
The most straightforward way would be to simply add this to sampling:
~~~python
df = df.sample(frac=1)
~~~
Which leaves it up to chance to balance it reasonably.

**Comment by AndreasKarasenko** [comment]
A forward search yields [this](https://link.springer.com/chapter/10.1007/978-3-031-56060-6_24) paper from 2024 which supports your last point and also points to [this](https://arxiv.org/pdf/2104.08786) paper from 2021/2022. You're probably right, that accounting for all biases might be out of scope. Maybe a best practices section would be appropriate then?

**Comment by OKUA1** [comment]
Yes, I agree that it is a good idea to at least mention it somewhere and in the future think about refactoring the code a bit to minimize this bias. 

I will keep the issue open for now.
